symphony-bdk-core/src/main/java/com/symphony/bdk/core/activity/ActivityRegistry.java

```diff
@@ -56,7 +56,8 @@ private void preProcessActivity(AbstractActivity<?, ?> activity) {
       Optional<AbstractActivity<?, ?>> act = this.activityList.stream()
```

Comment:
What we could do is: remove the `instanceof` check, implement an `equals` method in the `ActivityInfo` and have a unique key field in the Activity Info (which will uniquely identify the activity).